diff --git a/node_modules/fastest-validator/dist/index.d.ts b/node_modules/fastest-validator/dist/index.d.ts
index c005775..297ec72 100644
--- a/node_modules/fastest-validator/dist/index.d.ts
+++ b/node_modules/fastest-validator/dist/index.d.ts
@@ -317,7 +317,7 @@ export interface RuleObject extends RuleCustom {
 	 * if true any properties which are not defined on the schema will throw an error.
 	 * @default false
 	 */
-	strict?: boolean;
+	strict?: boolean | 'remove';
 	/**
 	 * List of properties that should be validated by this rule
 	 */
@@ -808,7 +808,7 @@ export type ValidationSchema<T = any> = {
 		/**
 		 * List of validation rules for each defined field
 		 */
-		[key in keyof T]: ValidationRule | undefined | any;
+		[key in keyof T]: ValidationRule | undefined;
 	};
 
 /**
